org.lflang/src/org/lflang/generator/c/CGenerator.xtend

```diff
@@ -1368,98 +1363,32 @@ class CGenerator extends GeneratorBase {
      * @param the content of the "services" section of the docker-compose.yml file.
      * @param the name of the federate to be added to "services".
      * @param the name of the federate's Dockerfile.
-     * @param the federationID.
      */
-    def appendFederateToDockerComposeServices(StringBuilder dockerComposeServices, String federateName, String dockerFileName, int dockerComposeFederationID) {
+    def appendFederateToDockerComposeServices(StringBuilder dockerComposeServices, String federateName, String context, String rtiName, String dockerFileName) {
```

Comment:
I think this might be as good an opportunity as any to put these functions in a separate `CDockerGenerator.java` file.